src/System.IO.Compression/src/System/IO/Compression/DeflateManaged/InflaterManaged.cs

```diff
@@ -105,7 +108,26 @@ public int Inflate(byte[] bytes, int offset, int length)
             int count = 0;
             do
             {
-                int copied = _output.CopyTo(bytes, offset, length);
+                int copied;
+                if (_uncompressedSize == -1)
+                {
+                    copied = _output.CopyTo(bytes, offset, length);
+                }
+                else
+                {
+                    if (_uncompressedSize > _currentInflatedCount)
+                    {
+                        copied = _output.CopyTo(bytes, offset, length);
+                        copied = Math.Min(copied, (int)(_uncompressedSize - _currentInflatedCount));
+                        _currentInflatedCount += copied;
+                    }
+                    else
+                    {
+                        copied = 0;
+                        _state = InflaterState.Done;
+                        _output._bytesUsed = 0;
```

Comment:
Why is it important to set _bytesUsed to 0?

Reply:
not that important, though there is an assert expecting all bytes will be used when inflation done, which should be true in normal cases except this one . I preferred to set it to 0 instead of removing that Assert: https://github.com/dotnet/corefx/pull/38314/files#diff-2b00ab942e3ce94f59570dddcf4693b6L152